tests/PHPStan/Reflection/Constant/RuntimeConstantReflectionTest.php

```diff
@@ -0,0 +1,59 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Reflection\Constant;
+
+use PhpParser\Node\Name;
+use PHPStan\Testing\PHPStanTestCase;
+use PHPStan\TrinaryLogic;
+use const PHP_VERSION_ID;
+
+class RuntimeConstantReflectionTest extends PHPStanTestCase
+{
+
+	public function dataDeprecatedConstants(): iterable
+	{
+		if (PHP_VERSION_ID >= 80100) {
+			yield [
+				new Name('\FILTER_SANITIZE_STRING'),
+				TrinaryLogic::createYes(),
+				'8.1', // deprecation message used in e.g. https://github.com/JetBrains/phpstorm-stubs/blob/9608c953230b08f07b703ecfe459cc58d5421437/filter/filter.php#L478
```

Comment:
Would be great to get rid of it, right. If the message === X.Y or X.Y.Z, we can remove it using regex

Reply:
filtered it out using regex. looking into the CI errors
https://github.com/phpstan/phpstan-src/actions/runs/8152892522/job/22283354469?pr=2953

I got the idea, we could also turn the deprecation-message into `Use of constant %s is deprecated since PHP %s.` instead of removing it completly. wdyt?

Reply:
>  we could also turn the deprecation-message into Use of constant %s is deprecated since PHP %s. instead of removing it completly. wdyt?

We don't do this for any other feature, I don't think it's worth it. Passing different `phpVersion` or simply analysing it on phpstan.org/try makes it obvious what's going on.